### DISCUSSION TITLE
Scikit-build Developer Meeting #7b: 2023-06-02

### DISCUSSION THREAD

**Discussion by henryiii** [scikit-build/scikit-build#992] [Community meeting notes]
Some topics to discuss at the meeting:

- Scientific Python Development Pages (+cookie, repo-review)
    - Planning to add a binary build page
- pyproject-validate supports plugins
    - Scikit-build-core could export its schema - could write converter for it
- Nanobind example & nanobind docs
    - Stuck on Win ARM, is linking to wrong lib (may have a fix)
- Adding specialty OSs to CI
- Asked about SciPy tools updates
- Should look into FindPython fully specified bug.

Scikit-build-core

- Planning to add `--strip` (maybe default conditional on minimum version?), `--component` support - could this avoid custom target support?
- Fixing editable installs on multi generators
- We could support multi-generators, actually, py-build-cmake has a detailed debugging story based on them
- Overrides are probably pretty easy to add now, but with `cp12` working, much less need
- GitHub CS shows 61 non-fork results for scikit-build-core, up from ~30 last time

**Comment by thewtex** [comment]
## Meeting notes

# Scikit-build Developer Meeting #7b: 2023-06-02

See https://github.com/orgs/scikit-build/discussions/992

## Updates

> - [Scientific Python Development Pages](https://learn.scientific-python.org/development) (+cookie, repo-review)
>     - Planning to add a binary build page

* https://github.com/scientific-python/repo-review
* https://github.com/scientific-python/cookie

<img width="50%" alt="example with scikit-build/scikit-build" src="https://hackmd.io/_uploads/rJZXLqwI3.png"/>

> - [pyproject-validate](https://pypi.org/project/validate-pyproject/) supports plugins
>     - Scikit-build-core could export its schema - could write converter for it


> - Nanobind example & nanobind docs
>     - Stuck on Win ARM, is linking to wrong lib (may have a fix)

* https://github.com/wjakob/nanobind
* https://nanobind.readthedocs.io/en/latest/
* https://github.com/wjakob/nanobind_example


> - Adding specialty OSs to CI

- https://github.com/vmactions

> - Asked about SciPy tools updates


> - Should look into FindPython fully specified bug.


### Scikit-build-core

> - Planning to add `--strip` (maybe default conditional on minimum version?), `--component` support - could this avoid custom target support?


> - Fixing editable installs on multi generators


> - We could support multi-generators, actually, py-build-cmake has a detailed debugging story based on them


> - Overrides are probably pretty easy to add now, but with `cp312` working, much less need


> - GitHub CS shows 61 non-fork results for scikit-build-core, up from ~30 last time

> CMake: Nice to have

Pyodide:

* Support setting CMAKE_CROSSCOMPILING_TOOLCHAIN as env. variable
    * https://github.com/emscripten-core/emscripten/pull/19301

Components:

* Support passing multiple components from the command line

**Reply by jcfr** [reply]
> Support setting CMAKE_CROSSCOMPILING_TOOLCHAIN as env. variable 

Follow-up contribution to CMake by @thewtex. See https://gitlab.kitware.com/cmake/cmake/-/merge_requests/8543
